src/Validation/Validator.php

```diff
@@ -3119,8 +3148,8 @@ public function getNotEmptyMessage(string $field): ?string
         }
 
         foreach ($this->_fields[$field] as $rule) {
-            if ($rule->get('rule') === 'notBlank' && $rule->get('message')) {
-                return $rule->get('message');
+            if ($rule->name === 'notBlank' && $rule->message) {
+                return $rule->message;
```

Comment:
I kind of like this readonly property setup. It exposes roughly the same public API that methods do but with less syntax.

How difficult would it be to support aliasing with this pattern? For example, we want to rename or deprecate a property, how would that work? Would we use `__get()`? or use a property hook?

Reply:
Either a virtual property getter or `__get()` could be used.